### PR TITLE
JBPM-6102: Stunner - Multiple selection/operations support

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis @hasys 

This is the stuff for the selection multiple support. See details at [JBPM-6102](https://issues.jboss.org/browse/JBPM-6102).

It **depends on** https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/629

Note: Notice Case Management is partially working - moving tasks inside same stage is not working fine at this point. Sorry about this, I realized about a bug after asking for the new lienzo release. I finally fixed the bug in Lienzo and asked for a new release, but unfortunately as you know releases are taking long time to be done... so I prefer to move all this work forward and once having the new release, hope soon tbh, will fix this issue. I've added a couple of `TODO` in `CaseManagementShapeControl` and `CaseManagementContainmentControl` for this. If you feel brave, you can locally build the lienzo-core upstream and uncomment these both `TODO`'s, and everything is working fine.. sorry again

### CODE REVIEW

**Comment by romartin** [comment]
@hasys another note for your testings - will notice that in case for an embedded subprocess, even selecting two nodes, which are connected by a sequence flow, and trying to add them inside an embedded subprocess, it's still complaining about the connector is invalid. This is not related with this selection multiple stuff in fact,, this is about core stuff (commands/rules), let you know that actually @tiagodolphine is working on it. Thanks!

**Comment by jomarko** [comment]
@romartin hi, I see that this PR affects the DMN authoring, would you mind incorporate me among reviewers in future for such PRs?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/session/DMNClientFullSessionTest.java [hunk 1]
```diff
@@ -72,7 +72,7 @@
     private PanControl<AbstractCanvas> panControl;
 
     @Mock
-    private DragControl<AbstractCanvasHandler, Element> dragControl;
+    private LocationControl<AbstractCanvasHandler, Element> dragControl;
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/session/DMNClientFullSessionTest.java]: @romartin please could we rename also the variable `dragControls` to `locationControls`?

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/ContainmentAcceptorControlImpl.java [hunk 2]
```diff
@@ -64,72 +66,98 @@ protected void onDisable(final WiresCanvas.View view) {
...
             return false;
         }
-        final CommandResult<CanvasViolation> result =
-                executor.apply(canvasCommandFactory.updateChildNode((Node) parent,
-                                                                    candidate));
-        if (highlightInvalid && CommandUtils.isError(result)) {
-            canvasHighlight.invalid(result.getViolations());
-        } else {
-            canvasHighlight.unhighLight();
+        // Do not accept multiple containment if candidates do not share same parent instance.
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/ContainmentAcceptorControlImpl.java]: As we renamed variable `candidate` to `children` could we use the name `children` also in the comment? Or is this naming intended?

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    @Override
+    public void enable(final AbstractCanvasHandler canvasHandler) {
+        super.enable(canvasHandler);
+        getWiresManager().setLocationAcceptor(LOCATION_ACCEPTOR);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void register(final Element element) {
+        if (null != element.asNode() && checkNotRegistered(element)) {
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java]: @romartin please, is the `null != element.asNode()` correct, I am asking because the `element.asNode()` is not called inside if the `if`or `checkNotRegistered`

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        if (elements.length != locations.length) {
+            throw new IllegalArgumentException("The length for the elements to move " +
+                                                       "does not match the locations provided.");
+        }
+        Command<AbstractCanvasHandler, CanvasViolation> command;
+        if (elements.length == 1) {
+            command = createMoveCommand(elements[0],
+                                        locations[0]);
+        } else {
+            final CompositeCommand.Builder<AbstractCanvasHandler, CanvasViolation> builder =
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java]: Not sure about performance, but seems to me that `CompositeCommand.Builder` for all cases when count of `elements` is bigger than 0.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+        public boolean allow(final WiresContainer[] wiresContainers,
+                             final com.ait.lienzo.client.core.types.Point2D[] point2DS) {
+            return true;
+        }
+
+        @Override
+        public boolean accept(final WiresContainer[] wiresContainers,
+                              final com.ait.lienzo.client.core.types.Point2D[] points) {
+            final Element[] elements = new Element[wiresContainers.length];
+            final Point2D[] locations = new Point2D[points.length];
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java]: Shouldn't we assert `wiresContainers` and `points` have same length? It will throw an `Exception` in the `for` below if `points` have smaller length than `wiresContainers`.

**Comment by hasys** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java (hunk 3 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java]: hey @jomarko , yeah I tihnk so. The checkNotRegistered method takes a generic 'Element' argument, which can be a Node or an Edge, because this method is shared between different controls. But for the specific case of the location control, it does not update connector's location, as in Stunner it's not allowed, at least for now, having "unconnected" connectors, so even if you drag and drop a connector and try to save, it will complain. So due to we don't need to store the connector's location (only we need the source/target magnets to the source/target shapes), there is no need to register it in this control.. Hope this makes sense for you! Thanks for the feedback!! :+1:

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java (hunk 5 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LocationControlImpl.java]: well yes and no.. this is Lienzo API - this method is a callback that you can use in Lienzo, so in fact it's the Lienzo implementation that calls this callbacks who has to ensure both arrays have same length. I mean we can add the check here, but if the check really happens, so if the array sizes are different, we cannot fix it either (in Stunner), as it's Lienzo who's sending the wrong arguments...  so tbh not sure if makes sense, although probably it's better doing the check and realizing about something is wrong rather than finally having some strange ArrayOutOfBounds exception... is it? :) Yeah adding it

**Comment by romartin** [comment]
Hey @jomarko 
Sorry for not adding you for review, will do from now on, np! I was not sure if you were involved on this now or not... I'm juast adding always @hasys. Anyway as more people reviewing much better yeah :)
Should I add you on all reviews? or just for the DMN specific ones? 
Thanks!

**Comment by romartin** [comment]
@jomarko added a new commit about your review's comments. Thanks!

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Hey @manstis @hasys @jomarko 
Finally the Lienzo version `2.0.294` has been released, so I'd been able to add a new commit which removes some `TODO` and cleans a bit more the code... Added a couple of missing tests as well. 
Thanks!

**Comment by jomarko** [comment]
@romartin thank you for responses to my comments, about reviews, yes feel free to add me to all PRs, however I will concentrate more on those affecting DMN Editor somehow.

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @hasys , please when you have a change, although full downstream build is not success, the artifacts have been generated, so you should be able to do your review... thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/main/java/org/kie/workbench/common/stunner/lienzo/core/shape/wires/WiresDragConstraintEnforcer.java [hunk 6]
```diff
@@ -26,6 +26,8 @@
 import com.ait.lienzo.client.widget.DragConstraintEnforcer;
 import com.ait.lienzo.client.widget.DragContext;
 
+// TODO: REMOVE!
+
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/main/java/org/kie/workbench/common/stunner/lienzo/core/shape/wires/WiresDragConstraintEnforcer.java]: It looks like this class is used only in itself and in test. Can we remove this class in this PR?

Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-lienzo-extensions/src/main/java/org/kie/workbench/common/stunner/lienzo/core/shape/wires/WiresDragConstraintEnforcer.java]: oh good point @hasys , thanks a lot, forgot to remove that class, is no longer necessary!! :+1:

**Comment by romartin** [comment]
@hasys thanks for the review, great job! Trying to fix your comments, will let you know ASAP. 
BTW in your comments you are referencing some pictures (pic1, pic2, etc), but I think you forgot to add the screenshots, can you add them please? Just to be sure we're talking about same thigns.. :)
Thanks

**Comment by romartin** [comment]
@hasys adding more commits but still not completed, no need for review yet, I'll let you know once done... Thanks!

**Comment by romartin** [comment]
Hey @hasys 

Really good catches, lots of thanks, really great job!! 

I have added several commits that hope are fixing (almost) all bugs from your comments above¡, just missing one to fix, the one about text wrapping, it's because there is no relationship with these commits and because I think this will be fixed as soon we can merge this old PR https://github.com/kiegroup/kie-wb-common/pull/1216, so we'll address it later. (as that PR depends on this one at same time).

Please give it a try and lemme know something, I'll rebase, squash and commit once you're happy.

Thanks!!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@jomarko missing from you to approve this PR as well (I mean click on the approve button), once you're happy. Thanks!!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @jomarko 

Oh good catch as well! The issue is due because of the use of a custom line implementation that exist in DMN (DirectionalLine), which was missing to re-parse the attributes after refreshing. So just fixed it and added some tests as well.

CC @hasys Need review from you too plz

Thanks again for your feedback and reviews guys!!

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Jenkins please retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonMarshaller.java [hunk 7]
```diff
@@ -2870,7 +2870,9 @@ private void correctEventNodeSize(BPMNShape shape) {
 //                  bounds.setY(y);
 //             }
         } else if (element instanceof Gateway) {
-            Bounds bounds = shape.getBounds();
+            // This statement does not apply in Stunner.
```
[the remark this answers is not in the document]
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonMarshaller.java]: hey @hasys , well not an issue anwyay but if you realize instead of removing this stuff I added a comment about Stunner... this is because these are the marshallers from old designer, and those are gonna be replaced (task starts next month!), so all these classes will be removed and this way once doing the new implementation we can see the "differences" between what stunner is doing and was designer was doing (at least about marshalling models).

**Comment by romartin** [comment]
Hey @hasys @jomarko @manstis 

Okk soo after applying most of the fixes for the bugs you found (thanks!!), here are a few comments:
- The fixes for the bugs that are on lienzo side have been fixed in [this development branch](https://github.com/romartin/lienzo-core/tree/wires-fixings-for-20294). Once you're happy with this PR, I'll create the PR for lienzo and ask for a new `2.0.295` release
- I've been able to fix most of the fixes in Stunner anyway by using the actual lienzo `2.0.294` release and overriding the classes in Stunner codebase. This way at least, until the next lienzo release is being pusblished, Stunner is as most stable as possible. 
- I've been not able to fix the issues about the crazy things that does the rectangle selection shape area in Stunner sorry, there are no extension points for this. But I've fixed it in lienzo anyway, so it will be fixed in Stunner once moving to next release.

This way I have created this JIRA ticket about missing to upgrade to next Lienzo version in order to fix some issues -> see [JBPM-6700](https://issues.jboss.org/browse/JBPM-6700)

Created as well a few tickets for issues found (not related):
- [JBPM-6701](https://issues.jboss.org/browse/JBPM-6701)
- [JBPM-6702](https://issues.jboss.org/browse/JBPM-6702)

On the other hand, no additional commits required, so once you all are happy, will rebase, squash and merge. 

@manstis Please I think it's still missing your review... thanks!

Thanks for all you help guys!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Just rebased from master and squashed commits - once build is success will merge, as this has been already approved.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/command/impl/ClearSessionCommand.java [hunk 8]
```diff
@@ -1,63 +0,0 @@
...
-import org.kie.workbench.common.stunner.core.client.canvas.AbstractCanvasHandler;
-import org.kie.workbench.common.stunner.core.client.canvas.event.command.CanvasCommandExecutedEvent;
-import org.kie.workbench.common.stunner.core.client.canvas.event.command.CanvasUndoCommandExecutedEvent;
-import org.kie.workbench.common.stunner.core.client.command.CanvasCommandFactory;
-import org.kie.workbench.common.stunner.core.client.command.SessionCommandManager;
-import org.kie.workbench.common.stunner.core.client.session.Session;
-
-@DMNEditor
-@Dependent
-public class ClearSessionCommand extends org.kie.workbench.common.stunner.core.client.session.command.impl.ClearSessionCommand {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/command/impl/ClearSessionCommand.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/command/impl/SessionCommandFactory.java [hunk 9]
```diff
@@ -1,77 +0,0 @@
...
-import org.kie.workbench.common.stunner.core.client.session.command.impl.ExportToPngSessionCommand;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.RedoSessionCommand;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.SwitchGridSessionCommand;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.UndoSessionCommand;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.ValidateSessionCommand;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.VisitGraphSessionCommand;
-
-@DMNEditor
-@ApplicationScoped
-public class SessionCommandFactory extends org.kie.workbench.common.stunner.core.client.session.command.impl.SessionCommandFactory {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/command/impl/SessionCommandFactory.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/command/ClearToolbarCommand.java [hunk 10]
```diff
@@ -1,34 +0,0 @@
...
-import javax.enterprise.context.Dependent;
-import javax.inject.Inject;
-
-import org.kie.workbench.common.dmn.api.qualifiers.DMNEditor;
-import org.kie.workbench.common.stunner.core.client.i18n.ClientTranslationService;
-import org.kie.workbench.common.stunner.core.client.session.command.impl.SessionCommandFactory;
-
-@DMNEditor
-@Dependent
-public class ClearToolbarCommand extends org.kie.workbench.common.stunner.client.widgets.toolbar.command.ClearToolbarCommand {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/command/ClearToolbarCommand.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/command/ToolbarCommandFactory.java [hunk 11]
```diff
@@ -1,77 +0,0 @@
...
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.PasteToolbarCommand;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.RedoToolbarCommand;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.SwitchGridToolbarCommand;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.UndoToolbarCommand;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.ValidateToolbarCommand;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.VisitGraphToolbarCommand;
-
-@DMNEditor
-@Dependent
-public class ToolbarCommandFactory extends org.kie.workbench.common.stunner.client.widgets.toolbar.command.ToolbarCommandFactory {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/command/ToolbarCommandFactory.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/impl/EditorToolbarFactory.java [hunk 12]
```diff
@@ -1,42 +0,0 @@
...
-import org.kie.workbench.common.dmn.api.qualifiers.DMNEditor;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.ToolbarView;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.command.ToolbarCommandFactory;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.impl.AbstractToolbar;
-import org.kie.workbench.common.stunner.client.widgets.toolbar.item.AbstractToolbarItem;
-import org.kie.workbench.common.stunner.core.client.session.impl.AbstractClientFullSession;
-
-@DMNEditor
-@ApplicationScoped
-public class EditorToolbarFactory extends org.kie.workbench.common.stunner.client.widgets.toolbar.impl.EditorToolbarFactory {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/toolbar/impl/EditorToolbarFactory.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/session/command/impl/ClearSessionCommandTest.java [hunk 13]
```diff
@@ -1,81 +0,0 @@
...
-import org.kie.workbench.common.stunner.core.client.canvas.AbstractCanvasHandler;
-import org.kie.workbench.common.stunner.core.client.command.CanvasCommandFactory;
-import org.kie.workbench.common.stunner.core.client.command.SessionCommandManager;
-import org.mockito.Mock;
-import org.mockito.runners.MockitoJUnitRunner;
-
-import static org.mockito.Mockito.*;
-
-@RunWith(MockitoJUnitRunner.class)
-public class ClearSessionCommandTest extends BaseCommandsTest {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/session/command/impl/ClearSessionCommandTest.java]: Why did you delete? This implementation had specific behaviour for the DMN Editor. Please re-instate.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/SessionDiagramEditorScreen.java [hunk 14]
```diff
@@ -113,7 +112,7 @@ public SessionDiagramEditorScreen(final DefinitionManager definitionManager,
                                       final ShowcaseDiagramService diagramService,
                                       final SessionManager sessionManager,
                                       final @Session SessionCommandManager<AbstractCanvasHandler> sessionCommandManager,
-                                      final @DMNEditor SessionPresenterFactory<Diagram, AbstractClientReadOnlySession, AbstractClientFullSession> sessionPresenterFactory,
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/SessionDiagramEditorScreen.java]: Why did you delete? The DMN Editor needed to use it's own implementation of ```SessionPresenterFactory```. Please re-instate the old behaviour and related classes this PR removes..

**Comment by manstis** [comment]
Happy New Year.... to find things broken :-(

I also have missing ```ToolboxAction``` for my nodes... <sigh>.. dig, dig, dig..

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNClientFullSession.java [hunk 15]
```diff
@@ -49,14 +48,14 @@
...
 
-    private DragControl<AbstractCanvasHandler, Element> dragControl;
+    private LocationControl<AbstractCanvasHandler, Element> locationControl;
     private ResizeControl<AbstractCanvasHandler, Element> resizeControl;
     private CanvasInPlaceTextEditorControl<AbstractCanvasHandler, AbstractClientFullSession, Element> canvasInPlaceTextEditorControl;
     private ToolboxControl<AbstractCanvasHandler, Element> toolboxControl;
 
     @Inject
     @SuppressWarnings("unchecked")
-    public DMNClientFullSession(final @DMNEditor CanvasFactory<AbstractCanvas, AbstractCanvasHandler> factory,
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNClientFullSession.java]: Please re-instate use of DMN's ```CanvasFactory``` as it did specific things required by the DMN Editor.

**Comment by romartin** [comment]
Hey @manstis , I'm back! :) Sorry not sure if I have to take some action here, did you fixed the classes I removed by error? should I add them? it's something missing at that point? Thanks!

**Comment by romartin** [comment]
@manstis and sorry for the inconveniences again

**Comment by manstis** [comment]
@romartin There's nothing to do now :-)
